src/libaktualizr/storage/sqlstorage.cc

```diff
@@ -1108,7 +1108,11 @@ class SQLTargetRHandle : public StorageTargetRHandle {
 };
 
 std::unique_ptr<StorageTargetRHandle> SQLStorage::openTargetFile(const std::string& filename) {
-  return std::unique_ptr<StorageTargetRHandle>(new SQLTargetRHandle(*this, filename));
+  try {
+    return std::unique_ptr<StorageTargetRHandle>(new SQLTargetRHandle(*this, filename));
+  } catch (...) {
```

Comment:
It should at least log some part of the exception.
Also, this method is used internally and in existing code, it doesn't check for `nullptr` (relies on exception).

So we should revisit all calls and make them check for `nullptr`. Alternatively (I would prefer that actually), the exception catching could be in the new `Aktualizr::GetStoredTarget` method.

Reply:
Yes, I agree it would be better/easier. Thank you!